x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/__mocks__/utils.ts

```diff
@@ -41,50 +38,39 @@ export const getSimpleRuleWithId = (id = 'rule-1'): Partial<OutputRuleAlertRest>
 });
 
 /**
- * Given an array of rule_id strings this will return a ndjson buffer which is useful
- * for testing uploads.
- * @param ruleIds Array of strings of rule_ids
- * @param isNdjson Boolean to determine file extension
+ * Given an array of rules, builds an NDJSON string of rules
+ * as we might import/export
+ * @param rules Array of rule objects with which to generate rule JSON
  */
-export const getSimpleRuleAsMultipartContent = (ruleIds: string[], isNdjson = true): Buffer => {
-  const arrayOfRules = ruleIds.map(ruleId => {
-    const simpleRule = getSimpleRule(ruleId);
-    return JSON.stringify(simpleRule);
-  });
-  const stringOfRules = arrayOfRules.join('\r\n');
-
-  const resultingPayload =
-    `--${TEST_BOUNDARY}\r\n` +
-    `Content-Disposition: form-data; name="file"; filename="rules.${
-      isNdjson ? 'ndjson' : 'json'
-    }\r\n` +
-    'Content-Type: application/octet-stream\r\n' +
-    '\r\n' +
-    `${stringOfRules}\r\n` +
-    `--${TEST_BOUNDARY}--\r\n`;
+export const rulesToNdJsonString = (rules: Array<Partial<OutputRuleAlertRest>>) => {
+  return rules.map(rule => JSON.stringify(rule)).join('\r\n');
+};
 
-  return Buffer.from(resultingPayload);
+/**
+ * Given an array of rule IDs, builds an NDJSON string of rules
+ * as we might import/export
+ * @param ruleIds Array of ruleIds with which to generate rule JSON
+ */
+export const ruleIdsToNdJsonString = (ruleIds: string[]) => {
+  const rules = ruleIds.map(ruleId => getSimpleRule(ruleId));
+  return rulesToNdJsonString(rules);
 };
 
 /**
- * Given an array of rule_id strings this will return a ndjson buffer which is useful
- * for testing uploads.
- * @param count Number of rules to generate
- * @param isNdjson Boolean to determine file extension
+ * Given a string, builds a hapi stream as our
+ * route handler would receive it.
+ * @param string contents of the stream
+ * @param filename String to declare file extension
  */
-export const getSimpleRuleAsMultipartContentNoRuleId = (count: number, isNdjson = true): Buffer => {
-  const arrayOfRules = Array(count).fill(JSON.stringify(getSimpleRuleWithId()));
-  const stringOfRules = arrayOfRules.join('\r\n');
+export const buildHapiStream = (string: string, filename = 'file.ndjson'): HapiReadableStream => {
+  const HapiStream = class extends Readable {
+    public hapi!: { filename: string };
```

Comment:
Here you can play by normal TypeScript rules and I think that would look better and keep us from copying examples of turning off TypeScript assertions through bang `!` low to non-existent:

```
/**
 * Given a string, builds a hapi stream as our
 * route handler would receive it.
 * @param string contents of the stream
 * @param filename String to declare file extension
 */
export const buildHapiStream = (string: string, filename = 'file.ndjson'): HapiReadableStream => {
  const HapiStream = class extends Readable {
    public hapi: { filename: string };
    constructor(fileName: string) {
      super();
      this.hapi = { filename: fileName };
    }
  };

  const stream = new HapiStream(filename);
  stream.push(string);
  stream.push(null);

  return stream;
};
```

That would construct a normal object which extends Readable and adds a fileName like you want.

Reply:
@FrankHassanabad I was avoiding defining a constructor since I didn't want to either break the existing constructor nor reimplement its behavior. Which do you think is the lesser of two evils here?

Reply:
Since it's for tests I think defining the constructor is ok. If you want to be super correct/strict you can define it to take options and pass those in along with the file name as a second argument:
https://nodejs.org/api/stream.html

<img width="1737" alt="Screen Shot 2020-03-03 at 1 21 41 PM" src="https://user-images.githubusercontent.com/1151048/75816222-ff965d80-5d51-11ea-93cf-42dfcc9bc6a3.png">

But I would just do it like you have it and call it a day since you already instantiate it and give it default options when you call super() like that.

Reply:
9ac14aefd61b1d0fb34ff326428f28d134fe6df7